### DISCUSSION TITLE
Why do we dont have a nullable array?

### DISCUSSION THREAD

**Discussion by sandbox-apps** [prisma/prisma#8618] [Q&A] [answered]
Why do we dont have a nullable array? Sometimes its useful to declare a array as null rather than empty specially when you have a Single Table Inheritance in which some property must be null because it doesnt apply to current subtype

**Comment by ryands17** [comment, marked as answer]
We haven't found any specific use case for having the many side of relations as nullable.

> Sometimes its useful to declare a array as null rather than empty specially when you have a Single Table Inheritance in which some property must be null because it doesnt apply to current subtype

This can be achieved with empty arrays as well.

**Reply by sandbox-apps** [reply]
But I think this violates Impossible states should be impossible. Where in my case I serialize the object and dont return the null value in api response but it is being returned because it has a value of empty. But ofcourse I can map it to a new object before returning but I just guess maybe its possible not to

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
